### PR TITLE
Followup after moving task property exploration around

### PR DESCRIPTION
**Pull request by lptr**
Improvements based on review feedback after https://github.com/gradle/gradle/pull/22739.

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: subprojects/model-core/src/main/java/org/gradle/internal/properties/bean/DefaultPropertyWalker.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 public class DefaultPropertyWalker implements PropertyWalker {
     private final RuntimeBeanNodeFactory nodeFactory;
 
-    public DefaultPropertyWalker(TypeMetadataStore typeMetadataStore, ImplementationIdentifier implementationIdentifier) {
+    public DefaultPropertyWalker(TypeMetadataStore typeMetadataStore, ImplementationResolver implementationIdentifier) {
```
Comment (wolfs) [inline subprojects/model-core/src/main/java/org/gradle/internal/properties/bean/DefaultPropertyWalker.java:31]:
```suggestion
    public DefaultPropertyWalker(TypeMetadataStore typeMetadataStore, ImplementationResolver implementationResolver) {
```

:x:

File: subprojects/model-core/src/main/java/org/gradle/internal/properties/bean/ImplementationResolver.java [hunk 2]
```diff
@@ -18,6 +18,6 @@
 
 import org.gradle.internal.snapshot.impl.ImplementationValue;
 
-public interface ImplementationIdentifier {
-    ImplementationValue identify(Object bean);
+public interface ImplementationResolver {
```
[the remark this answers is not in the document]
Comment (lptr) [reply subprojects/model-core/src/main/java/org/gradle/internal/properties/bean/ImplementationResolver.java:21]: I'm planning to change this, too, so let's just keep it as is for now.

File: subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/AbstractTaskInputsAndOutputsTest.groovy [hunk 3]
```diff
@@ -92,14 +92,14 @@ abstract class AbstractTaskInputsAndOutputsTest extends AbstractProjectBuilderSp
...
             void visitInputFileProperty(
-                String propertyName,
-                boolean optional,
-                InputBehavior behavior,
-                DirectorySensitivity emptyDirectorySensitivity,
-                LineEndingSensitivity lineEndingNormalization,
-                @Nullable Normalizer fileNormalizer,
-                PropertyValue value,
-                InputFilePropertyType filePropertyType
+                    String propertyName,
```
[the remark this answers is not in the document]
Comment (lptr) [reply subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/AbstractTaskInputsAndOutputsTest.groovy:95]: Reverted.

File: subprojects/model-core/src/main/java/org/gradle/internal/properties/bean/DefaultPropertyWalker.java (hunk 1 above)
Comment (lptr) [reply subprojects/model-core/src/main/java/org/gradle/internal/properties/bean/DefaultPropertyWalker.java]: Fixed.

**Review by wolfs — APPROVED** [review]

**Comment by lptr** [comment]
@bot-gradle test and merge please.

**Comment by lptr** [comment]
@bot-gradle test and merge please.

**Comment by lptr** [comment]
@bot-gradle test and merge please.

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**